actors.go

```diff
@@ -7,9 +7,22 @@ import (
 	"os/signal"
 )
 
+// ContextHandler returns an actor, i.e. an execute and interrupt func, that
+// terminates with when the parent context is canceled.
+func ContextHandler(ctx context.Context) (execute func() error, interrupt func(error)) {
+	ctx, cancel := context.WithCancel(ctx)
+	return func() error {
+			<-ctx.Done()
+			return ctx.Err()
+		}, func(error) {
+			cancel()
+		}
+}
```

Comment:
I don't totally grok the use case, here. In my experience, any time you create a run.Group that would be subject to a context like the ctx here, (nearly) every actor in that group would take a context derived from ctx. And as long as one actor is stopped by that context being canceled, that's enough. 

But I guess you've got some function that takes a context, and create a run.Group with a bunch of actors, but none of those actors have execute functions that take/respond-to a context? Is that correct?

Reply:
```go
func (a App) Run(ctx context.Context) error {
	rg := run.Group{}

	rg.Add(a.grpcHandler.Start,
		func(_ error) {
			a.grpcHandler.GracefulStop()
		})

	rg.Add(a.httpHandler.Start,
		func(_ error) {
			if err := a.httpHandler.GracefulStop(); err != nil {
				a.log.Error(err, "http metric server shutdown error")
			}
		})

	rg.Add(run.SignalHandler(ctx))

	return rg.Run()
```

Yep, this is totally correct. This is my particular case that I encountered. I'm not entirely sure if it wouldn't be better to create some kind of a wrapper function for grpcHandler.GracefulStop and httpHandler.GracefulStop that would take a context and have a select statement inside. I originally did it as shown in the code snippet because I read doc-string of `SignalHandler` and it seemed to me that it could be used in my particular case.

I understand if you don't want to make the pretty minimalistic library any bigger. I just thought that if the SignalHandler was supposed to work as written in the documentation, you assumed that someone could use it with only context. So, I don't mind if you just clarify the SignalHandler docs and leave out ContextHandler addition.

Anyway thank you for the library and attention to this PR.

Reply:
No, I think ContextHandler may make sense, I just have to think about it a little bit! Thanks for the contribution.